public/app/plugins/panel/traces/module.tsx

```diff
@@ -1,6 +1,105 @@
-import { PanelPlugin } from '@grafana/data';
+import { PanelPlugin, toOption } from '@grafana/data';
 
+import { getTraceServiceNames, getTraceSpanNames } from '../../../features/explore/TraceView/utils/tags';
+import { transformDataFrames } from '../../../features/explore/TraceView/utils/transform';
+
+import { TagsEditor } from './TagsEditor';
 import { TracesPanel } from './TracesPanel';
 import { TracesSuggestionsSupplier } from './suggestions';
 
-export const plugin = new PanelPlugin(TracesPanel).setSuggestionsSupplier(new TracesSuggestionsSupplier());
+export const plugin = new PanelPlugin(TracesPanel)
+  .setPanelOptions((builder, context) => {
+    const category = ['Span filters'];
+    const trace = transformDataFrames(context.data[0]);
```

Comment:
Should fail gracefully rather than crash if there is no data.

Reply:
I think `context.data` should always be defined but can be empty, but `transformDataFrames()` would return `null` [in that case](https://github.com/grafana/grafana/blob/37dbf037de548e1e855c0a59697769d20a8b5b65/public/app/features/explore/TraceView/utils/transform.ts#L5) so it shouldn't crash